File: lib/core/config/backend_config.dart
/// Compile-time backend URL when [.env] is unavailable (native killed-state
/// HTTP, cold start before prefs are cached).
///
/// Set via `--dart-define=API_BASE_URL=...` (mirrored in Android
/// [BuildConfig.API_BASE_URL]). Keep in sync with `.env` at dev time.
const String kDefaultProductionApiBaseUrl =
    String.fromEnvironment('API_BASE_URL');

/// SharedPreferences key (no `flutter.` prefix — Dart plugin adds it on Android).
const String kNativeApiBaseUrlPrefsKey = 'api_base_url';

File: android/app/src/main/kotlin/com/munawwaracare/android/BackendConfig.kt
package com.munawwaracare.android

/**
 * Backend URL for native code paths when Flutter SharedPreferences are empty
 * (cold start / killed app before prefs are cached from Dart).
 *
 * Value comes from `--dart-define=API_BASE_URL=...` at build time
 * ([BuildConfig.API_BASE_URL]). Keep in sync with `.env` and
 * `lib/core/config/backend_config.dart`.
 */
object BackendConfig {
    val API_BASE_URL_FALLBACK: String
        get() = BuildConfig.API_BASE_URL
}
